cpp/include/cudf/strings/string_view.cuh

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstddef>
+#include <cuda_runtime.h>
+#include <cudf/types.hpp>
+
+namespace cudf
+{
+
+// UTF-8 characters are 1-4 bytes
+using char_utf8 = uint32_t;
+
+/**---------------------------------------------------------------------------*
+ * @brief A non-owning, immutable view of device data that is variable length
```

Comment:
```suggestion
 * @brief A non-owning, immutable view of device data that is a variable length
```

Reply:
That does not sound right to me.

Reply:
data that is variable length array /
data that is a variable length array

Not really important

Reply:
You are right. I did not read the whole thing.